explore/src/main/scala/explore/config/ConfigurationPanel.scala

```diff
@@ -100,70 +101,70 @@ object ConfigurationPanel {
-          SpectroscopyConfigurationPanel(spectroscopy.as(dataIso))
-            .when(isSpectroscopy),
-          ImagingConfigurationPanel(imaging)
-            .unless(isSpectroscopy)
-        ),
-        SpectroscopyModesTable
-          .component(
-            SpectroscopyModesTable(
-              configurationView,
-              props.ctx.staticData.spectroscopyMatrix,
-              spectroscopy.get
-            )
+  private def renderFn(
+    props:           Props,
+    scienceDataUndo: UndoContext[ScienceData],
+    state:           StateSnapshot[State]
+  )(implicit ctx:    AppContextIO): VdomNode = {
+    val requirementsCtx = scienceDataUndo.zoom(ScienceData.requirements)
+
+    val requirementsViewSet = UndoView(props.obsId, requirementsCtx)
+
+    def mode           = requirementsViewSet(ScienceRequirementsData.mode, UpdateScienceRequirements.mode)
+    val isSpectroscopy = mode.get === ScienceMode.Spectroscopy
+
+    val spectroscopy = requirementsViewSet(
+      ScienceRequirementsData.spectroscopyRequirements,
+      UpdateScienceRequirements.spectroscopyRequirements
+    )
+
+    val imaging = state.zoomStateL(State.imagingOptions)
+
+    val configurationView = scienceDataUndo
+      .undoableView(ScienceData.configuration)
+      .withOnMod(conf => setScienceConfiguration(props.obsId, conf).runAsync)
+
+    <.div(
+      ExploreStyles.ConfigurationGrid,
+      props.renderInTitle(<.span(ExploreStyles.TitleUndoButtons)(UndoButtons(scienceDataUndo))),
+      Form(size = Small)(
+        ExploreStyles.Grid,
+        ExploreStyles.Compact,
+        ExploreStyles.ExploreForm,
+        ExploreStyles.ConfigurationForm
+      )(
+        <.label("Mode", HelpIcon("configuration/mode.md")),
+        EnumViewSelect(id = "configuration-mode", value = mode),
+        SpectroscopyConfigurationPanel(spectroscopy.as(dataIso))
+          .when(isSpectroscopy),
+        ImagingConfigurationPanel(imaging)
+          .unless(isSpectroscopy)
+      ),
+      SpectroscopyModesTable
+        .component(
+          SpectroscopyModesTable(
+            configurationView,
+            props.ctx.staticData.spectroscopyMatrix,
+            spectroscopy.get
           )
-          .when(isSpectroscopy)
-      )
-    }
-
-    def render(props: Props) = AppCtx.using { implicit appCtx =>
-      renderFn(
-        props,
-        props.scienceDataUndo
-      )
-    }
+        )
+        .when(isSpectroscopy)
+    )
   }
 
   protected val component =
-    ScalaComponent
-      .builder[Props]
-      .initialState(State(ScienceMode.Spectroscopy, ImagingConfigurationOptions.Default))
-      .renderBackend[Backend]
-      .configure(Reusability.shouldComponentUpdate)
-      .build
+    ScalaFnComponent
+      .withHooks[Props]
+      .useStateSnapshotWithReuse(
+        State(ScienceMode.Spectroscopy, ImagingConfigurationOptions.Default)
+      )
+      .render { (props, state) =>
+        AppCtx.using { implicit appCtx =>
```

Comment:
We already have the context in `props.ctx`, no need to go get it from the `AppCtx` object. We can either `implicit val ctx = props.ctx` or pass it explicitly to `renderFn`.

(This could have been optimized like in the class component version too).

Reply:
an nice